fix: fixes transactionID isolation for GraphQL

diesieben07 opened the pull request:
## Description

Fixes #4091 by using a JavaScript Proxy to isolate the `transactionID` between individual GraphQL operations instead of copying the request.
I have also fixed the missing cloning of the request for the `collections/findByID` resolver.

- [X] I have read and understand the [CONTRIBUTING.md](https://github.com/payloadcms/payload/blob/main/CONTRIBUTING.md) document in this repository.

## Type of change

<!-- Please delete options that are not relevant. -->

- [X] Bug fix (non-breaking change which fixes an issue)

## Checklist:

- [X] I have added tests that prove my fix is effective or that my feature works
- [X] Existing test suite passes locally with my changes

DanRibbens commented on packages/payload/src/utilities/isolateTransactionID.ts:
```diff
@@ -0,0 +1,23 @@
+/**
+ * Creates a proxy for the given request that has its own TransactionID
+ */
+export default function isolateTransactionID(req: PayloadRequest): PayloadRequest {
```
I was looking for a way to do this and couldn't find it. Great work!

DanRibbens reviewed (changes requested):
If you do a search for `req: { ...context.req } as PayloadRequest` you'll find a few more places where I did this which your PR should also replace.

diesieben07 commented:
@DanRibbens Right, of course :man_facepalming: I've now gone through `registerSchema` and checked every resolver.

Should we expose `isolateTransactionID` for people to use in custom GraphQL resolvers or maybe we should handle it transparently by wrapping any custom resolver with it? The latter sounds more convenient for users and makes it impossible to forget.

DanRibbens commented:
> Should we expose `isolateTransactionID` for people to use in custom GraphQL resolvers or maybe we should handle it transparently by wrapping any custom resolver with it? The latter sounds more convenient for users and makes it impossible to forget.

We never want people to have to learn these kinds of ugly details if it can be avoided. I haven't reviewed this yet but it sounds like we need to handle it.

Thanks!

diesieben07 commented:
I have now added a wrapper to any (top level) resolver in the custom mutations and queries. I've added a test that makes sure they are using separate transaction IDs.
